File: 1. Study/0610 - 자료구조(스택,큐)/mojaeya/2_쇠막대기.js
console.log(solution("()(((()())(())()))(())"));

function solution(arr) {
  let answer = 0;
  let stack = [];
  for (let i = 0; i < arr.length; i++) {
    if (x === "(") stack.push(arr[i]);
    else {
      stack.pop();
      if (arr[i - 1] === "(") answer += stack.length; //레이저
      else answer += 1; // 막대기 끝
    }
  }
}

File: 1. Study/0610 - 자료구조(스택,큐)/kimshin/2. 쇠막대기.py
a=input()

stack=[]
cnt=0
for x in range(len(a)):
    if a[x] == '(':
        stack.append(a[x])
    else:
        if a[x-1] == '(':
            stack.pop()
            cnt+=len(stack)
        else:
            cnt+=1
print(cnt)